dali/base/dasds.cpp

```diff
@@ -1407,17 +1427,36 @@ class CDeltaWriter : implements IThreaded
         CriticalBlock b(pendingCrit);
         addToQueue(item);
     }
-    void clear()
+    void flush()
     {
-        CriticalBlock b(pendingCrit);
-        pending = {};
+        bool waitOnEmptySem = false;
+        {
+            CriticalBlock b(pendingCrit);
+            if (waiting)
+            {
+                if (pendingSz)
+                    signalWaiting();
+            }
+            if (!waiting)
+            {
+                signalEmpty = true;
+                waitOnEmptySem = true;
+            }
+        }
+        if (waitOnEmptySem)
```

Comment:
"allWritten" would be a bit clearer than "empty"

Reply:
agree, will rename this and some others to clarify meaning.